Loop/AppDelegate.swift

```diff
@@ -27,62 +27,108 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, DeviceOrientationCo
+            let fileURL = documentDirectory.appendingPathComponent("protection.test")
+            guard fileManager.fileExists(atPath: fileURL.path) else {
+                let contents = Data("unimportant".utf8)
+                try? contents.write(to: fileURL, options: .completeFileProtectionUntilFirstUserAuthentication)
+                // If file doesn't exist, we're at first start, which will be user directed.
+                return true
+            }
+            let contents = try? Data(contentsOf: fileURL)
+            return contents != nil
+        } catch {
+            log.error("Could not create after first unlock test file: %@", String(describing: error))
+        }
+        return false
+    }
+    
+    private func finishLaunch(application: UIApplication) {
+        log.default("Finishing launching")
         UIDevice.current.isBatteryMonitoringEnabled = true
 
         bluetoothStateManager = BluetoothStateManager()
-        bluetoothStateManager.addBluetoothStateObserver(rootViewController.rootViewController)
         alertManager = AlertManager(rootViewController: rootViewController, expireAfter: Bundle.main.localCacheDuration ?? .days(1))
         deviceDataManager = DeviceDataManager(pluginManager: pluginManager, alertManager: alertManager, bluetoothStateManager: bluetoothStateManager, rootViewController: rootViewController)
+
+        let statusTableViewController = UIStoryboard(name: "Main", bundle: Bundle(for: AppDelegate.self)).instantiateViewController(withIdentifier: "MainStatusViewController") as! StatusTableViewController
+        
+        statusTableViewController.deviceManager = deviceDataManager
+
+        bluetoothStateManager.addBluetoothStateObserver(statusTableViewController)
+
         loopAlertsManager = LoopAlertsManager(alertManager: alertManager, bluetoothStateManager: bluetoothStateManager)
         
         SharedLogging.instance = deviceDataManager.loggingServicesManager
 
+        deviceDataManager?.analyticsServicesManager.application(application, didFinishLaunchingWithOptions: launchOptions)
+
         OrientationLock.deviceOrientationController = self
 
         NotificationManager.authorize(delegate: self)
-        
-        log.info(#function)
 
-        deviceDataManager.analyticsServicesManager.application(application, didFinishLaunchingWithOptions: launchOptions)
 
-        rootViewController.rootViewController.deviceManager = deviceDataManager
-        
+        rootViewController.pushViewController(statusTableViewController, animated: false)
+
         let notificationOption = launchOptions?[.remoteNotification]
         
         if let notification = notificationOption as? [String: AnyObject] {
-            deviceDataManager.handleRemoteNotification(notification)
+            deviceDataManager?.handleRemoteNotification(notification)
         }
+    }
```

Comment:
Do we want to nil out `launchOptions` when `finishLaunching` is complete? Otherwise it is holding on to it for the lifetime of the app run.